Purchases: Extract PurchaseMeta component

ryelle opened the pull request:
This PR creates a new component for the meta info on a purchase detail page (price, expires on, payment method) and adds a new meta detail, "owner". The styles have also been updated to match the mockups in #12746 

Jetpack site:
<img width="733" alt="jetpack" src="https://cloud.githubusercontent.com/assets/541093/25013030/38c515e4-2040-11e7-81f6-d9d9efa569b1.png">

Domain included in plan:
<img width="744" alt="domain-with-plan" src="https://cloud.githubusercontent.com/assets/541093/25013031/38c583b2-2040-11e7-9885-b6e7cbd34e0a.png">

A disconnected site
<img width="740" alt="disconnected-site" src="https://cloud.githubusercontent.com/assets/541093/25013029/38c4594c-2040-11e7-9368-5132f1ee920d.png">

Other than adding the Owner section, there should be no functionality change in the other sections. I've created this component to simplify ManagePurchase.

A basic test — make sure nothing looks broken:

- Check your purchase list view, /me/purchases
- Check a wp.com site
- Check a Jetpack site

To test in-depth, check various plan combos:

- wp.com site, jetpack site, or disconnected jetpack site
- plan expired, plan expiring within 90 days, plan not expiring/auto-renew enabled
- credit card expiring before plan renews

**Note**: this is currently compared to add/purchase-site-header-component, #13003 — once that's merged I'll switch this back to master.

rickybanister commented:
This looks beautiful. I have only one microscopic nitpic—can we just link the word 'plan' in the domain mapping upgrade, rather than have 'view plan' beneath it? That will tighten up and simplify that layout. 

The borders, typography, spacing of everything else looks great—did you reuse styles from /stats?

rickybanister commented:
We could probably also just say `Subscription renews on` instead of `auto-renews`. I think auto-renew is a sort of made-up a8c word.

ryelle commented:
I rebased this PR with master now that the Header PR is merged. I also updated the auto-renews wording & moved the plan link to the "Renews with plan" text:

<img width="748" alt="screen shot 2017-04-13 at 5 55 05 pm" src="https://cloud.githubusercontent.com/assets/541093/25025973/736f69ce-2072-11e7-9429-13e79986d783.png">

a8ci18n commented on client/me/purchases/manage-purchase/purchase-meta.jsx:
```diff
@@ -0,0 +1,345 @@
+		}
+
+		if ( isSubscription( purchase ) ) {
+			return translate( 'Subscription renews on' );
```
Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Subscription auto-renews on' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/me/purchases/manage-purchase/purchase-meta.jsx:
```diff
@@ -0,0 +1,345 @@
+		}
+
+		if ( isDomainRegistration( purchase ) ) {
+			return translate( 'Domain renews on' );
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Domain auto-renews on' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

rickybanister commented:
That's much better @ryelle. This looks ready to merge from my perspective.
